api/paidAction/itemCreate.js

```diff
@@ -12,8 +12,32 @@ export const paymentMethods = [
 ]
 
 export async function getCost ({ subName, parentId, uploadIds, boost = 0, bio }, { models, me }) {
-  const sub = (parentId || bio) ? null : await models.sub.findUnique({ where: { name: subName } })
-  const baseCost = sub ? satsToMsats(sub.baseCost) : 1000n
+  const BIO_COST = 1000n
+  async function findRootItem (models, parentId) {
+    const item = await models.item.findFirst({
+      where: { id: Number(parentId) },
+      select: { rootId: true }
+    })
+
+    return models.item.findFirst({
+      where: { id: Number(item.rootId || parentId) },
+      include: { sub: true }
+    })
+  }
+  async function getBaseCost ({ models, bio, parentId, subName }) {
+    if (bio) return BIO_COST
+
+    if (parentId) {
+      const rootItem = await findRootItem(models, parentId)
+      return rootItem.bio ? BIO_COST : satsToMsats(rootItem.sub?.replyCost)
+    }
+
+    const sub = await models.sub.findUnique({ where: { name: subName } })
+    return satsToMsats(sub.baseCost)
+  }
```

Comment:
possibly opinionated nitpick: I think it's easier to read if these functions are defined in the top level (with space between them :eyes:). As written, it suggests there is a "bigger reason" to be defined in the scope of the function but the functions take everything they need as arguments anyway (except `BIO_COST` but that can then also be defined at the top level) so there's also some shadowing going on.

Reply:
Opinionated or not, to have these functions as reusable (and readable!) is a good thing!